src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -60,7 +60,9 @@ public ListPanel(UI ui, IModel model, PanelControl parentPanelControl, int panel
      * @return true if the issue has changed, false otherwise
      */
     private boolean issueHasNewComments(TurboIssue issue, boolean hasMetadata) {
-        if (currentFilterExpression.getQualifierNames().contains(Qualifier.UPDATED) && hasMetadata) {
+        List<String> filterQualifierNames = currentFilterExpression.getQualifierNames();
+        if ((filterQualifierNames.contains(Qualifier.UPDATED) ||
```

Comment:
Do we need to check for the self update qualifier as well?

`hasMetadata` could be put first the boolean expression. The presence of qualifiers can be checked into a static `Qualifier` helper method.